### DISCUSSION TITLE
Calculating String Order parameters

### DISCUSSION THREAD

**Discussion by ArnabeshS** [QuSpin/QuSpin#535] [General]
Is there a way to calculate [string order parameters](https://iopscience.iop.org/article/10.1088/1361-648X/ab41b5) in Quspin? I have a one to calculate given below:
![string_order](https://user-images.githubusercontent.com/34402914/169049939-20aaf418-8126-44ba-8d0a-54157735bb44.png)

My idea is to construct an object of the `hamiltonian` class for this operator and then take the expectation against the ground state of a model. But constructing the `static` list seems to be a problem for me. The list for local order parameters like sigma_z can be constructed simply, but for this case, the size of each small list grows. How this problem can be tackled?

Here is my take:
For constructing one element of the `static` list, I've written the following program:
```
L=8
operator_list=[]
            L1=1
            while(L1<L):
                i=0.0
                y=[1.0]
                op_list=[]
                w=""
                while(i<L1):
                    y.append(float(i))
                    w+='x'
                    if(i==L1-1):
                        y.append(i%L1)
                        y.append((i+1)%L1)
                        w+="yz"
                        op_list.append(w)
                        op_list.append(y)
                    i+=1.0
                operator_list.append(op_list)
                L1+=1
```
However, setting the `shape=(2**L,2**L)`  as the parameter in hamiltonian I am getting the following error:
```
[['xyz', [1.0, 0.0, 0.0, 0.0]], ['xxyz', [1.0, 0.0, 1.0, 1.0, 0.0]], ['xxxyz', [1.0, 0.0, 1.0, 2.0, 2.0, 0.0]]]
Traceback (most recent call last):

  File "C:\Users\arnabesh\Desktop\order_parameter-3\order_param_1.3_final.py", line 79, in <module>
    M2 = hamiltonian([operator_list],[],**kwargs)

  File "C:\Users\arnabesh\anaconda3\lib\site-packages\quspin\operators\hamiltonian_core.py", line 376, in __init__
    O = _np.asarray(O,dtype=self._dtype)

  File "C:\Users\arnabesh\anaconda3\lib\site-packages\numpy\core\_asarray.py", line 102, in asarray
    return array(a, dtype, copy=False, order=order)

ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (3, 2) + inhomogeneous part.
```
Can anyone throw some light on this?

**Comment by weinbe58** [comment]
this is almost correct, your output should look like:

`[['xyz', [[1.0, 0.0, 0.0, 0.0]]], ['xxyz', [[1.0, 0.0, 1.0, 1.0, 0.0]]], ['xxxyz', [[1.0, 0.0, 1.0, 2.0, 2.0, 0.0]]]]`
